lib/Basics/StringUtils.cpp

```diff
@@ -1406,6 +1414,119 @@ std::string soundex(char const* src, size_t const len) {
   return result;
 }
     
+    
+template<typename InputType>
+inline bool eq(InputType const& c1, InputType const& c2) {
+    return c1 == c2;
+}
+
+template<typename InputType, typename LengthType>
+LengthType impl(InputType const* lhs,
```

Comment:
`impl` should be renamed here, as it is now part of a StringUtils library with multiple functions.
Additionally, both `eq` and `impl` should be placed in the anonymous namespace.